add glossary entry for 'implementation' (#277)
json-schema-org/website: pages/learn/glossary.md

The [JSON Schema specification](../specification) defines behavior for a large library of keywords which can be used to describe [instances](#instance).

### implementation

> Historically, the word "implementation" has been used to describe both specifically validators/annotators (interpreted as strictly operating as the spec describes) and also various other kinds of JSON Schema tooling.  However, due to this ambiguity, we have decided that the preferred conversational term should be ["tooling"](#tooling).

Reviewer: ```suggestion
Historically, the word "implementation" has been used to describe both specifically validators/annotators and also various other kinds of JSON Schema tooling.  However, due to this ambiguity, we have decided that the preferred conversational term should be ["tooling"](#tooling).
```

Author: I think that note is important for context.

Reviewer: I'm not suggesting the historical note be removed. My concern is that this sounds like something is not considered an implementation if it doesn't strictly conform to the spec. It sounds like ajv wouldn't be considered an implementation, which isn't accurate, even historically.

Author: Okay.  The note you're suggesting to remove is intended to express that the spec only covers validation/annotation.  What would you have to express that?

Reviewer: My best suggestion might be "(which is what the spec describes)". The words feel a little awkward, but I think it conveys the intended meaning.

Author: How now?

Reviewer: Perfect :+1:

The [JSON Schema specification](../specification) defines behavior for a large library of keywords which can be used to describe [instances](#instance).

### implementation

> An implementation is any piece of software that provides or exposes behavior defined by the JSON Schema specification.

Reviewer: Taking a stab at addressing @Relequestual's concerns about the wording here...

```suggestion
An implementation is any software tooling or library for working with or evaluating schemas in some way that is not inconsistent with the JSON Schema specification.
```

Author: "not inconsistent"?

not just "consistent"?

Reviewer: I thought about that, but I thought it was more clear this way despite the double negative. I wanted to convey that an implementation might do more or less than what the spec says as long as as it doesn't contradict it. I felt like "consistent" could be interpreted as referring to the whole spec, which is not necessarily the case for all types of implementations.

But, I'm bad at words, so that might not make any sense.

Subschemas may themselves contain sub-subschemas, though colloquially one generally uses the term "subschema" regardless of the level of nesting, further clarifying which larger schema is the parent schema whenever needed.

### tooling

> A JSON Schema tool (or colloquially "tooling") is any software application or library for working with or evaluating schemas in some way that does not contradict the JSON Schema specification.

Reviewer: ```suggestion
A JSON Schema tool (or colloquially "tooling") is any software application or library for working with or evaluating schemas in some way.
```
